examples/hyperparameter-optimization.ipynb

```diff
@@ -82,81 +66,70 @@
     "X, y = boston.data, boston.target\n",
     "n_features = X.shape[1]\n",
     "\n",
-    "reg = GradientBoostingRegressor(n_estimators=50, random_state=0)\n",
-    "\n",
-    "def objective(params):\n",
-    "    max_depth, learning_rate, max_features, min_samples_split, min_samples_leaf = params\n",
-    "\n",
-    "    reg.set_params(max_depth=max_depth,\n",
-    "                   learning_rate=learning_rate,\n",
-    "                   max_features=max_features,\n",
-    "                   min_samples_split=min_samples_split, \n",
-    "                   min_samples_leaf=min_samples_leaf)\n",
-    "\n",
-    "    return -np.mean(cross_val_score(reg, X, y, cv=5, n_jobs=-1,\n",
-    "                                    scoring=\"neg_mean_absolute_error\"))"
+    "# gradient boosted trees tend to do well on problems like this\n",
+    "reg = GradientBoostingRegressor(n_estimators=50, random_state=0)"
    ]
   },
   {
    "cell_type": "markdown",
-   "metadata": {
-    "deletable": true,
-    "editable": true
-   },
+   "metadata": {},
    "source": [
-    "Next, we need to define the bounds of the dimensions of the search space we want to explore:"
+    "Next, we need to define the bounds of the dimensions of the search space we want to explore and pick the objective. In this case the cross-validation mean absolute error of a gradient boosting regressor over the Boston dataset, as a function of its hyper-parameters."
    ]
   },
   {
    "cell_type": "code",
    "execution_count": 3,
-   "metadata": {
-    "collapsed": false,
-    "deletable": true,
-    "editable": true
-   },
+   "metadata": {},
    "outputs": [],
    "source": [
-    "space  = [(1, 5),                           # max_depth\n",
-    "          (10**-5, 10**0, \"log-uniform\"),   # learning_rate\n",
-    "          (1, n_features),                  # max_features\n",
-    "          (2, 100),                         # min_samples_split\n",
-    "          (1, 100)]                         # min_samples_leaf"
+    "from skopt import space as sk_space\n",
+    "from skopt.utils import use_named_args\n",
+    "\n",
+    "\n",
+    "# The list of hyper-parameters we want to optimize. For each one we define the bounds,\n",
+    "# the corresponding scikit-learn parameter name, as well as how to sample values\n",
+    "# from that dimension (`'log-uniform'` for the learning rate)\n",
+    "space  = [sk_space.Integer(1, 5, name='max_depth'),\n",
+    "          sk_space.Real(10**-5, 10**0, \"log-uniform\", name='learning_rate'),\n",
+    "          sk_space.Integer(1, n_features, name='max_features'),\n",
+    "          sk_space.Integer(2, 100, name='min_samples_split'),\n",
+    "          sk_space.Integer(1, 100, name='min_samples_leaf')]\n",
```

Comment:
`from skopt.space import Real, Integer`? that would remove some of the boilerplate.

Reply:
👍